include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl.h

```diff
@@ -1929,6 +1951,30 @@ __parallel_find_or(oneapi::dpl::__internal::__device_backend_tag, _ExecutionPoli
         return __result != __init_value ? __result : __rng_n;
 }
 
+template <typename _ExecutionPolicy, typename _Brick, typename _BrickTag, typename... _Ranges>
+::std::conditional_t<
+    ::std::is_same_v<_BrickTag, __parallel_or_tag>, bool,
+    oneapi::dpl::__internal::__difference_t<typename oneapi::dpl::__ranges::__get_first_range_type<_Ranges...>::type>>
+__parallel_find_or(oneapi::dpl::__internal::__device_backend_tag, _ExecutionPolicy&& __exec, _Brick __f,
+                   _BrickTag __brick_tag, _Ranges&&... __rngs)
+{
+    return __parallel_find_or_impl(oneapi::dpl::__internal::__device_backend_tag{},
+        std::forward<_ExecutionPolicy>(__exec), __f, __brick_tag, __first_size_calc<_Ranges...>{},
+        std::forward<_Ranges>(__rngs)...);
+}
+
+template <typename _ExecutionPolicy, typename _Brick, typename _BrickTag, typename... _Ranges>
+::std::conditional_t<
+    ::std::is_same_v<_BrickTag, __parallel_or_tag>, bool,
+    oneapi::dpl::__internal::__difference_t<typename oneapi::dpl::__ranges::__get_first_range_type<_Ranges...>::type>>
+__parallel_find_or_min_size(oneapi::dpl::__internal::__device_backend_tag, _ExecutionPolicy&& __exec, _Brick __f,
```

Comment:
Was it necessary to introduce this distinction?

I am not totally sure, but it seem the original `__parallel_find_or` assumed the same size of all the ranges. If so, we could keep it as it is, do not introduce `__parallel_find_or_min_size` with new `__parallel_find_or`, and adjust the sizes of the input range before calling the original `__parallel_find_or` to the common minimal size.

Reply:
Actually, there is not good way to adjust the sizes of the input ranges before calling the pattern `__parallel_find_or_impl`...
Only, wrapping the input ranges into `take_view `or `subrange`, for example. It is not good approach because we are changing the original type. 
In `__parallel_find_or_impl` pattern we are operating already with one-dimensional iteration space  - integer value "n".. and we can just take "n" as minimum of size of the input ranges.